src/classes/AffiliationMappingsHealthCheckVMapper.cls

```diff
@@ -198,6 +257,325 @@ public virtual with sharing class AffiliationMappingsHealthCheckVMapper {
+    * @return A HealthCheckGroupVModel
+    **********************************************************************************/ 
+    @TestVisible 
+    private virtual HealthCheckGroupVModel getValidContactPrimaryFieldHealthCheckGroupVModel(
+        AffiliationMappingsModel affiliationMappingsModel,
+        Map<String, LookupFieldModel> accountlookupFieldOnContactFieldModelByName,
+        Map<String, LookupFieldModel> accountlookupFieldOnContactFieldModelByLabel
+    ) {
+        List<HealthCheckItemVModel> healthCheckItemsList = new List<HealthCheckItemVModel>();
+        String healthCheckGroupLabel = Label.stgHealthCheckAffiliationMappings;
+
+        HealthCheckGroupVModel healthCheckGroupVModel = new HealthCheckGroupVModel(healthCheckGroupLabel,
+                                                                                 Label.stgHealthCheckStatusPassed,
+                                                                                 0,
+                                                                                 0,
+                                                                                 new List<String>(),
+                                                                                 healthCheckItemsList);
+
+        LookupFieldModel accountlookupFieldOnContactFieldModelFromName = this.locateLookupFieldService().getLookupFieldModelFromNameMap(
+            affiliationMappingsModel.primaryAffiliationField, 
+            accountlookupFieldOnContactFieldModelByName);
+
+        if (accountlookupFieldOnContactFieldModelFromName == null) {
+            LookupFieldModel accountlookupFieldOnContactFieldModelFromLabel = this.locateLookupFieldService().getLookupFieldModelFromLabelMap(
+                affiliationMappingsModel.primaryAffiliationField,
+                accountlookupFieldOnContactFieldModelByLabel);
+            
+            if (accountlookupFieldOnContactFieldModelFromLabel == null) {
+                healthCheckGroupVModel.healthCheckItemList.add(new HealthCheckItemVModel(
+                    '03',
+                    Label.stgHCAfflMapContactPrimaryFieldTitle,
+                    'Fail',
+                    Label.stgHealthCheckStatusFailed,
+                    String.format(
+                        Label.stgHCAfflMapContactPrimaryFieldInvalid, 
+                        new List<Object>{affiliationMappingsModel.primaryAffiliationField}
+                    ),
+                    String.format(
+                        Label.stgHCAfflMapContPrimaryFieldInvalidFix, 
+                        new List<Object>{affiliationMappingsModel.primaryAffiliationField}
+                    ),
+                    new List<HealthCheckItemVModel>()
+                ));
+
+                healthCheckGroupVModel.totalChecks++;
+                healthCheckGroupVModel.updateHealthCheckStatus();
+                
+                return healthCheckGroupVModel;
+            }
+        }
```

Comment:
We can isolate the if statement here and avoid nesting. This also can help readability as it becomes more clear that we are defaulting over to the other contact field model.

Another option that I prefer would be to make a helper method just like in the record type service where we get the lookupfieldmodel by the two maps. This might make code readability even better and isolate the get by name or label behavior within the service class. When we start settings work, we may rely on that method in the service class, so doing so will be defensive design.

```suggestion
        if (accountlookupFieldOnContactFieldModelFromName == null) {
            LookupFieldModel accountlookupFieldOnContactFieldModelFromLabel = this.locateLookupFieldService().getLookupFieldModelFromLabelMap(
                affiliationMappingsModel.primaryAffiliationField,
                accountlookupFieldOnContactFieldModelByLabel);
        }
            
        if (accountlookupFieldOnContactFieldModelFromLabel == null) {
            healthCheckGroupVModel.healthCheckItemList.add(new HealthCheckItemVModel(
                '03',
                Label.stgHCAfflMapContactPrimaryFieldTitle,
                'Fail',
                Label.stgHealthCheckStatusFailed,
                String.format(
                    Label.stgHCAfflMapContactPrimaryFieldInvalid, 
                    new List<Object>{affiliationMappingsModel.primaryAffiliationField}
                ),
                String.format(
                    Label.stgHCAfflMapContPrimaryFieldInvalidFix, 
                    new List<Object>{affiliationMappingsModel.primaryAffiliationField}
                ),
                new List<HealthCheckItemVModel>()
            ));

            healthCheckGroupVModel.totalChecks++;
            healthCheckGroupVModel.updateHealthCheckStatus();
            
            return healthCheckGroupVModel;
        }
```